rtl/eclrtl/eclregex.cpp

```diff
@@ -407,7 +407,22 @@ class CCompiledStrRegExpr final : implements ICompiledStrRegExpr
             failWithPCRE2Error(errNum, "Error in regex pattern: ", std::string(_regex, _regexLength), errOffset);
         }
 
-        compiledRegex = std::shared_ptr<pcre2_code_8>(newCompiledRegex, pcre2_code_free_8);
+        return newCompiledRegex;
+    }
+
+public:
+    CCompiledStrRegExpr(int _regexLength, const char * _regex, bool _isCaseSensitive, bool _enableUTF8, bool _throwOnError = false)
+    : isUTF8Enabled(_enableUTF8)
+    {
+        try
+        {
+            compiledRegex = std::shared_ptr<pcre2_code_8>(compileRegex(_regexLength, _regex, _isCaseSensitive, _enableUTF8), pcre2_code_free_8);
+        }
+        catch (IException *e)
```

Comment:
One alternative implementation suggestion.  You could have an `Owned<IException> savedException` member.

```
   catch (IException * e)
   {
      savedExceptions.setown(e);
   }
```
and
```
if (savedException)
    throw savedException.getLink();
```